sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1832,7 +1861,6 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                 }
 
                 partitionProcessor.Dispose();
-                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
```

Comment:
This is somewhat orthogonal, but it should never have been here.   At this point, the partition processor has been stopped.  Honoring the token here just caused the handlers not to be invoked, violating their contract.